server/src/librelay/relay_server.go

```diff
@@ -460,13 +461,60 @@ func (relay *RelayServer) CreateRelayTransaction(request RelayTransactionRequest
 		log.Println(err, request.RelayMaxNonce)
 		return
 	}
+	// canRelay returned true, so we can relay the tx
+	relayAddress := relay.Address()
+
+	callOpt := &bind.CallOpts{
+		From:    relayAddress,
+		Pending: false,
+	}
+
+	// With a transition to sponsor-defined gas limits, the server will need to crunch some numbers
+	sponsor, err := testcontracts.NewTestSponsor(request.GasSponsor, relay.Client)
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	gasLimits, err := sponsor.GetGasLimitsForSponsorCalls(&bind.CallOpts{From: relayAddress})
+	if err != nil {
+		log.Println(err)
+		return
+	}
+
+	hubOverhead, err := relay.rhub.GetHubOverhead(&bind.CallOpts{From: relayAddress})
+	if err != nil {
+		log.Println(err)
+		return
+	}
+	maxPossibleGas := big.NewInt(0)
+	maxPossibleGas.Add(maxPossibleGas, hubOverhead)
+	maxPossibleGas.Add(maxPossibleGas, gasLimits.AcceptRelayedCallGasLimit)
+	maxPossibleGas.Add(maxPossibleGas, gasLimits.PreRelayCallGasLimit)
+	maxPossibleGas.Add(maxPossibleGas, &request.GasLimit)
+	maxPossibleGas.Add(maxPossibleGas, gasLimits.PostRelayCallGasLimit)
+
+	// TODO: well, this sucks! I need to calculate max possible gas using entire msg.data!!!
+	//  In the meantime, these 400'000 extra gas should help :-)
+	maxPossibleGas.Add(maxPossibleGas, big.NewInt(400000))
```

Comment:
Not sure what's that

Reply:
Hack to pass a big enough 'maxPossibleGas' to the ARC method. What I want is
```relayHub.methods.relayCall(...).encodeABI().length()```
but I don't know how to achieve that in Go and it is too late to put effort to it.